common/domain/dlq_message_handler_test.go

```diff
@@ -102,6 +107,125 @@ func (s *dlqMessageHandlerSuite) TestReadMessages() {
 	s.Nil(token)
 }
 
+func (s *dlqMessageHandlerSuite) TestStart() {
+	tests := []struct {
+		name           string
+		initialStatus  int32
+		expectedStatus int32
+		shouldStart    bool
+	}{
+		{
+			name:           "Should start when initialized",
+			initialStatus:  common.DaemonStatusInitialized,
+			expectedStatus: common.DaemonStatusStarted,
+			shouldStart:    true,
+		},
+		{
+			name:           "Should not start when already started",
+			initialStatus:  common.DaemonStatusStarted,
+			expectedStatus: common.DaemonStatusStarted,
+			shouldStart:    false,
+		},
+	}
+
+	for _, test := range tests {
+		s.Run(test.name, func() {
+			atomic.StoreInt32(&s.dlqMessageHandler.status, test.initialStatus)
+			s.dlqMessageHandler.Start()
+			if test.shouldStart {
+				s.dlqMessageHandler.logger.Info("Domain DLQ handler started.")
+			}
+			s.Equal(test.expectedStatus, atomic.LoadInt32(&s.dlqMessageHandler.status))
+		})
+	}
+	s.dlqMessageHandler.Stop()
```

Comment:
this doesn't look right. it should be stopped within the context it is started. 
```
s.dlqMessageHandler.Start()
defer s.dlqMessageHandler.Stop()
```

Reply:
We can put it in shouldStart. But not defer for both cases, as in 2nd case (Should not start when already started) Daemon is already started. 
``	s.dlqMessageHandler.Start()
			s.Equal(test.expectedStatus, atomic.LoadInt32(&s.dlqMessageHandler.status))
			if test.shouldStart {
				s.dlqMessageHandler.logger.Info("Domain DLQ handler started.")
				s.dlqMessageHandler.Stop()
			}
			``

Reply:
or can do something like:
s.NotPanics(func() {
                s.dlqMessageHandler.Stop()
            })
for multiple stop calls irrespective of less number of start

Reply:
what is down side of `defer stop()`? it should work whether it's started or not

Reply:
No, it does not. stop is not able to complete because of panic:
panic: close of closed channel

Reply:
so, if you see the first statement in Start there it is checking whether it is already started or not, and if it was already started Daemon will be not be started again.

DaemonStart.

Reply:
It's because you are initializing `dlqMessageHandler` inside `SetupTest` and it gets reused by all test cases inside `TestStart`. I recommend not using the suite at all here. We want full isolation and control over object lifecycle in these tests.